src/lib/url.js

```diff
@@ -53,10 +56,64 @@ export const parseQueryParameterFamily = (baseName, query) => {
 export const parseDictionaryParameter = (baseName, query) => {
   const family = parseQueryParameterFamily(baseName, query);
   return Object.keys(family).reduce((dictionary, key) => {
-    return Object.assign({}, dictionary, {[key]: new Set(parseListParameter(family[key]))});
+    return Object.assign({}, dictionary, {
+      [key]: new Set(parseListParameter(family[key])),
+    });
   }, {});
 };
 
+export const compileListParameter = value => {
+  return [...value].join(',');
+};
+
+export const compileDictionaryParameter = (baseName, type, query) => {
+  return `${baseName}[${type}]=${compileListParameter(query[type])}`;
+};
+
+export const compileQueryParameterFamily = (baseName, query) => {
+  const merge = (a, b) => {
+    return `${a}=${encodeURIComponent(b)}`;
+  };
+
+  const compile = (current, param) => {
+    if (typeof param === 'object') {
+      for (let [key, value] of Object.entries(param)) {
+        current += `[${key}]`;
+        if (value instanceof Set) {
+          return [...value].map(val => `${current}[]=${val}`).join('&');
+        } else if (typeof value === 'string') {
+          return merge(current, value);
```

Comment:
In the case of a string, `merge` calls `encodeURIComponent`, but what if the values in a `Set` are strings. Should those be encoded too?

Actually, for my understanding, what's an example where encoding is necessary at all?

Reply:
The only case in the test suite was the `=` in operator -that had to be uri encoded. Also- yes, once it hits a string value it should probably be encoded.

Reply:
Now this looks good to me